### PR TITLE
Postfixes for #409 and #476 plus #473

### PR DESCRIPTION
**Pull request by drnikolaev**
Postfixes for #409 and #476: removed getcwd hacks, fixed local imports
#473 merged on top

### CODE REVIEW

File: tests/unit/core/test_deploy_export.py [hunk 1]
```diff
@@ -53,28 +54,52 @@
...
+    )
+    from .tensorrt_runner import TensorRTRunnerV2
 except:
     # Skip tests.
     pytestmark = pytest.mark.skip
 
 
 @pytest.mark.usefixtures("neural_factory")
 class TestDeployExport(TestCase):
+    def setUp(self):
```
Comment (tkornuta-nvidia) [inline tests/unit/core/test_deploy_export.py]: Hey, why you added this once again?

The "neural_factory" fixture that the class uses is creating the Neural Factory (self.nf) already. It also handles the device/placement...

Besides, the "run_only_on("GPU") pytest marker already makes sure that a given test will run only on GPU. So those lines are redundant...

Also, not sure about the logging level. When you use pytest, it captures the streams and not print anything unless you tell it specifically. So once again, setting logger level to WARNING (which will result in skipping INFO) will have no impact on that is shown.

In short, the whole setUp() seems to be redundant, please considering removing  it.

File: nemo/core/neural_modules.py [hunk 2]
```diff
@@ -392,6 +392,30 @@ def output_ports(self) -> Optional[Dict[str, NeuralType]]:
           A (dict) of module's output ports names to NeuralTypes mapping
         """
 
+    @property
```
Comment (tkornuta-nvidia) [inline nemo/core/neural_modules.py]: I like idea of moving the whole deployment to NeuralModule class, seems that having those methods here is the first step.

Having said that, I am not sure about the methods on them own, i.e. is disabling input/output ports for deployment is a _generic_ solution to all modules?

Besides, seems that you are changing the NM interface available to the user. Are you sure that we want to make them even "accessible"? In other words - I think those should be private methods of the class.

Comment (ryanleary) [reply nemo/core/neural_modules.py]: I'm not sure without significant refactoring we can achieve making these fully private. We can change them to `_disabled_deployment_input_ports`, `_disabled_deployment_output_ports`, and `_prepare_for_deployment` so they're not part of the public NM API easily enough. Would that be sufficient for now since we intend to rework this further in the next release?

File: nemo/backends/pytorch/actions.py [hunk 3]
```diff
@@ -951,11 +951,13 @@ def __extract_dynamic_axes(port_name: str, ntype: NeuralType, dynamic_axes: defa
         # for input_ports
         for port_name, ntype in module.input_ports.items():
             if port_name in inputs_to_drop:
+                input_names.remove(port_name)
```
Comment (ryanleary) [inline nemo/backends/pytorch/actions.py]: Other actions.py changes from #473 are missing.

File: nemo/core/neural_modules.py (hunk 2 above)
Comment (tkornuta-nvidia) [reply nemo/core/neural_modules.py]: I agree, makes sense for now.

My goal was to trigger the discussion, but it is clear that those changes shouldn't be part of the current release. Code freeze is not the right moment. That is why I am also quite surprised that during code freeze we change the public interface, understand what @okuchaiev is saying about "braking change" - but am still surprised... ;)

File: nemo/backends/pytorch/actions.py (hunk 3 above)
Comment (tkornuta-nvidia) [reply nemo/backends/pytorch/actions.py]: Ok, I do not know #473 but some things are now clearly redundant, at least when it comes to jasper deployment...
